src/metabase/lib/schema.cljc

```diff
@@ -21,14 +23,24 @@
    [:native any?]
    [:args {:optional true} [:sequential any?]]])
 
+(mr/def ::breakouts
+  [:sequential {:min 1} [:ref ::ref/ref]])
+
+(mr/def ::fields
+  [:sequential {:min 1} [:ref ::ref/ref]])
+
 (mr/def ::stage.mbql
   [:and
    [:map
-    [:lib/type [:= :mbql.stage/mbql]]
+    [:lib/type    [:= :mbql.stage/mbql]]
     [:lib/options ::common/options]
-    [:order-by {:optional true} ::order-by/order-bys]
-    [:joins {:optional true} ::join/joins]
-    [:filter {:optional true} ::expression/boolean]]
+    [:joins       {:optional true} [:ref ::join/joins]]
+    [:expressions {:optional true} [:ref ::expression/expressions]]
```

Comment:
I wrapped some of this stuff in `:ref` so changes to these other schemas in other namespaces will get reflected in anything using this schema without having to reload a bunch of things.